lib/spack/spack/solver/concretize.lp

```diff
@@ -494,12 +505,23 @@ requirement_weight(Package, Group, W) :-
   requirement_policy(Package, Group, "any_of"),
   requirement_group_satisfied(Package, Group).
 
-error(100, "Cannot satisfy the requirements in packages.yaml for package '{0}'. You may want to delete them to proceed with concretization. To check where the requirements are defined run 'spack config blame packages'", Package) :-
-  activate_requirement_rules(Package),
+error(100, "cannot satisfy a requirement for package '{0}'.", Package) :-
+  activate_requirement(Package, X),
   requirement_group(Package, X),
+  not requirement_message(Package, X, _),
   not requirement_group_satisfied(Package, X).
 
+
+error(10, Message) :-
```

Comment:
I kept this ordering because of the following rationale:
1. Any valid ASP solution that results in an error is a failed concretization. Said otherwise, we can't have any `error` with any weight for a successful concretization. This means that any weight will preserve the correctness of concretization.
2. In the presence of an error, I'd like to prefer reporting requirements with custom messages, rather than other possible causes (e.g. having multiple compilers on a node), since it's likely they're more comprehensible to users. The hope is that users could diagnose a failed concretization better this way. To prefer reporting these errors, we need to assign them a lower weight.
3. Currently we have three weights for errors, which are `1`, `10` and `100`. I didn't want to introduce a new one in this PR.